cmd/skopeo/sync.go

```diff
@@ -283,6 +289,36 @@ func imagesToCopyFromDir(dirPath string) ([]types.ImageReference, error) {
 	return sourceReferences, nil
 }
 
+// imagesToCopyFromOci builds a list of image references from the images found
+// in the source directory.
+// It returns an image reference slice with as many elements as the images found
+// and any error encountered.
+func imagesToCopyFromOci(dirPath string) ([]types.ImageReference, error) {
+
+	var sourceReferences []types.ImageReference
+
+	indexJSON, err := os.Open(path.Join(dirPath, "index.json"))
+	if err != nil {
+		return sourceReferences, errors.Wrapf(err, "Cannot open index.json for transport %q and Path: %q", oci.Transport.Name(), dirPath)
+	}
+	defer indexJSON.Close()
+
+	index := &imgspecv1.Index{}
+	if err := json.NewDecoder(indexJSON).Decode(index); err != nil {
+		return sourceReferences, errors.Wrapf(err, "Cannot decode index.json for transport %q and Path: %q", oci.Transport.Name(), dirPath)
+	}
+
+	for _, md := range index.Manifests {
+		ref, err := oci.NewReference(dirPath, md.Annotations[imgspecv1.AnnotationRefName])
```

Comment:
(`AnnotationRefName` can legitimately be not set. Also https://github.com/containers/common/blob/31e8cc4aeeabbb50538fe3948d1fa3d7506a733a/libimage/pull.go#L176 . Either way, this should be in c/image.)